### PR TITLE
Implemented GetUserByEmail() and GetUserByPhoneNumber()

### PR DESCRIPTION
**Pull request by hiranya911**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Unreleased
 
--
+- [added] Implemented the `GetUserByEmailAsync()` and `GetUserByPhoneNumberAsync()`
+  APIs in the `FirebaserAuth` class.
```
Comment (bklimt) [inline CHANGELOG.md]: s/FirebaserAuth/FirebaseAuth/

File: FirebaseAdmin/FirebaseAdmin/Auth/FirebaseUserManager.cs [hunk 2]
```diff
@@ -68,27 +73,68 @@ public static FirebaseUserManager Create(FirebaseApp app)
...
-                "accounts:lookup", payload, cancellationToken).ConfigureAwait(false);
-            if (response == null || response.Users == null || response.Users.Count == 0)
+        /// <summary>
+        /// Gets the user data corresponding to the given email address.
+        /// </summary>
+        /// <param name="email">An email address.</param>
+        /// <param name="cancellationToken">A cancellation token to monitor the asynchronous
+        /// operation.</param>
+        /// <returns>A record of user with the queried email if one exists.</returns>
+        internal async Task<UserRecord> GetUserByEmail(
```
Comment (bklimt) [inline FirebaseAdmin/FirebaseAdmin/Auth/FirebaseUserManager.cs]: This should be named GetUserByEmailAsync.

File: FirebaseAdmin/FirebaseAdmin/Auth/FirebaseUserManager.cs [hunk 3]
```diff
@@ -68,27 +73,68 @@ public static FirebaseUserManager Create(FirebaseApp app)
...
+        }
+
+        /// <summary>
+        /// Gets the user data corresponding to the given phone number.
+        /// </summary>
+        /// <param name="phoneNumber">A phone number.</param>
+        /// <param name="cancellationToken">A cancellation token to monitor the asynchronous
+        /// operation.</param>
+        /// <returns>A record of user with the queried phone number if one exists.</returns>
+        internal async Task<UserRecord> GetUserByPhoneNumber(
```
Comment (bklimt) [inline FirebaseAdmin/FirebaseAdmin/Auth/FirebaseUserManager.cs]: Ditto.

File: FirebaseAdmin/FirebaseAdmin/Auth/FirebaseUserManager.cs [hunk 4]
```diff
@@ -195,5 +248,36 @@ private TResult SafeDeserialize<TResult>(string json)
                 throw new FirebaseException("Error while calling Firebase Auth service", e);
             }
         }
+
+        private class UserQuery
```
Comment (bklimt) [inline FirebaseAdmin/FirebaseAdmin/Auth/FirebaseUserManager.cs]: Would be nice to have a top-level comment describing this class.

File: CHANGELOG.md (hunk 1 above)
Comment (hiranya911) [reply CHANGELOG.md]: Done

File: FirebaseAdmin/FirebaseAdmin/Auth/FirebaseUserManager.cs (hunk 2 above)
Comment (hiranya911) [reply FirebaseAdmin/FirebaseAdmin/Auth/FirebaseUserManager.cs]: Done

File: FirebaseAdmin/FirebaseAdmin/Auth/FirebaseUserManager.cs (hunk 3 above)
Comment (hiranya911) [reply FirebaseAdmin/FirebaseAdmin/Auth/FirebaseUserManager.cs]: Done

File: FirebaseAdmin/FirebaseAdmin/Auth/FirebaseUserManager.cs (hunk 4 above)
Comment (hiranya911) [reply FirebaseAdmin/FirebaseAdmin/Auth/FirebaseUserManager.cs]: Done

**Pull request merged**